xwiki-pro-macros-test/xwiki-pro-macros-test-docker/src/test/it/com/xwiki/pro/test/ui/GenericMacrosIT.java

```diff
@@ -132,5 +278,524 @@ void teamMacroTest(TestUtils setup)
+        assertFalse(page.isSubtle(0));
+        assertTrue(page.isSubtle(1));
+        assertFalse(page.isSubtle(2));
+    }
+
+    @Test
+    @Order(4)
+    void tagListMacroTest(TestUtils setup)
+    {
+        setup.deletePage(pageWithTagListMacros);
+        setup.gotoPage("XWiki", "TagListTest");
+        setup.createPage(pageWithTagListMacros, PAGE_WITH_TAGLIST_MACROS_CONTENT);
+
+        TagListMacroPage page = new TagListMacroPage();
+
+        // There should be 3 tagList macros.
+        assertEquals(3, page.getTagListCount());
+
+        //Checks the ordered titles of the tags from the tagList
+        List<String> expectedTitles1 = Arrays.asList("A-B", "G");
+        assertEquals(expectedTitles1, page.getGlossaryTitles(0));
+
+        //Checks the ordered titles of the tags from the tagList
+        List<String> expectedTitles2 = Arrays.asList("A-G");
+        assertEquals(expectedTitles2, page.getGlossaryTitles(1));
+
+        //Checks the ordered names of the tags from the tagList
+        List<String> expectedTagNames1 = Arrays.asList("alpha", "beta", "gamma");
+        assertEquals(expectedTagNames1, page.getTagNames(0));
+
+        //Checks the ordered names of the tags from the tagList
+        List<String> expectedTagNames2 = Arrays.asList("alpha", "gamma");
+        assertEquals(expectedTagNames2, page.getTagNames(1));
+
+        //Checks the <a> tags of the tags from the tagList
+        for (String i : expectedTagNames1) {
+            assertEquals("a", page.getHtmlTagForTagName(0, i));
+        }
+
+        //Checks the <a> tags of the tags from the tagList
+        for (String i : expectedTagNames2) {
+            assertEquals("a", page.getHtmlTagForTagName(1, i));
+        }
+
+        //Checks the ordered titles of the tags from the tagList, multiple spaces
+        List<String> expectedTitles3 = Arrays.asList("A-X", "Y-Z");
+        assertEquals(expectedTitles3, page.getGlossaryTitles(2));
+
+        //Checks the ordered names of the tags from the tagList
+        List<String> expectedTagNames3 = Arrays.asList("alpha", "beta", "testTag", "x", "y", "z");
+        assertEquals(expectedTagNames3, page.getTagNames(2));
+
+        //Checks the <a> tags of the tags from the tagList
+        for (String i : expectedTagNames3) {
+            assertEquals("a", page.getHtmlTagForTagName(2, i));
+        }
+    }
+
+    @Test
+    @Order(5)
+    void expandMacroTest(TestUtils setup)
+    {
+        setup.deletePage(pageWithExpandMacro);
+        setup.gotoPage("Main", "ExpandMacroTest");
+        setup.createPage(pageWithExpandMacro, PAGE_WITH_EXPAND_MACROS_CONTENT);
+
+        ExpandMacroPage page = new ExpandMacroPage();
+
+        // There should be 2 expand macros.
+        assertEquals(2, page.getExpandCount());
+        //Nested Expand macros - Checks the titles of the macros
+        assertEquals("ExpandTest1", page.getTitleText(0));
+        assertEquals("ExpandTest2", page.getTitleText(1));
+
+        //Checking the icon
```

Comment:
```suggestion
        // Checking the icon.
```

Comments should end with a `.`, and you need a space after `//`